graph/api/src/main/java/org/jboss/windup/graph/service/GraphService.java

```diff
@@ -280,6 +280,34 @@ protected T getUnique(Traversal<?, ?> query)
         return frame((Vertex)resultObj);
     }
 
+    public T getOrCreate(String key, String value, String... kvs)
```

Comment:
Since the method is based on properties and there's already a naming convention in place in this classe for `ByProperties` methods, I suggest to add such a prefix.
```suggestion
    public T getOrCreateByProperties(String key, String value, String... kvs)
```